### DISCUSSION TITLE
Possible to hide all business icons in iOS app?

### DISCUSSION THREAD

**Discussion by dickorydock** [organicmaps/organicmaps#6780] [Ideas]
I want to be able to see just the map but without icons for restaurants or businesses. I didn't see this as a feature in the app or as an existing feature request so I would like to begin the discussion of it here.

**Comment by biodranik** [comment]
Please describe your use case in details to better understand the motivation.

**Comment by dickorydock** [comment]
Sure and thank you for inquiring! My main issue is that sometimes street names don't show up. I've noticed that in an area where there are a lot of businesses, a street with many businesses often won't have its actual street name labelled anywhere on the map. My hypothesis is that Organic Maps tends to put street name labels on blocks with fewer marked items, which means that on streets that have lots of businesses, sometimes street names don't show up at all. This can make it harder to get around.

I'd like to be able to hide businesses in the hopes that then street names will show up. Alternately, I'd like to have the option of always prioritizing printing street names over business names. But really, having the options of hiding the icons altogether would be preferable. The way I imagine this working would be that you could toggle on/off the display of certain categories of icons ("business", "restaurant", "government building" or something like that). 

Please let me know if you have any further questions or clarifications about the motivation here.

**Reply by biodranik** [reply]
Should street visibility be fixed instead of hiding POIs? Does it work better in other maps?
CC @pastk

**Reply by dickorydock** [reply]
Prioritizing street visibility is the goal for sure. By other maps do you mean other map apps or...? OSM-based Magic Earth has the capability to hide whole categories and that fixes street visibility issues there. Companies like Google Maps with an advertising-based model, understandably, don't allow you to hide businesses and have similar street visibility issues.

**Reply by biodranik** [reply]
@dickorydock What are you doing when you are looking at the street names in OM? Do you plan a car drive? Do you plan where to go on foot/what to visit? Are you already in the town and looking at the names of nearby streets where you need to go? What is the core reason to look at street names with your eyes (instead of using the search, for example)?

**Reply by dickorydock** [reply]
@biodranik Could you please describe how that information from me will be useful so I'll better understand the motivation of the questions?

**Reply by biodranik** [reply]
Because there are many ways to solve the issue if all necessary details are known. For example, we're planning to introduce different map styles for different activities. You don't need to stick to what OM or other apps offer now. We are building a better future product by collecting real users' cases and needs.

**Reply by dickorydock** [reply]
Yeah, I like the idea of the Hiking map style I read about in a different GitHub issue (see my comment below). The commenter below also mentioned that something like this is actively in the works and I'm glad to hear it.  Yes it's important to imagine beyond the current features- that is indeed why I submitted this idea for a future feature! 

There are lots of ways I might use this feature. Maybe I'm making a screenshot of a location and I want it to include all the intersecting streets. Maybe I'm a tourist and someone suggested I visit nearby "Smythe Street near the park" but I don't know how to spell it- seeing it clearly and easily on the map will help me find it. I could go on. I think the commenters in #2355 also have stories to share.

**Reply by biodranik** [reply]
Thanks. Any real-life cases are very important to build a good product.

**Comment by pastk** [comment]
Related to
- #2355

There is also an idea to try to render street names "under" POIs. If a name is partly overlayed with 1-2 icons, then it might miss just a few letters and be recognisable / readable still. But it'll become messy if overlayed with another text (a POI label)..

The planned "Driving" map style will likely have more emphasis on street names.

**Reply by dickorydock** [reply]
Yeah, in the area I'm looking at there are just so many POIs that it would get quite messy.

Thanks for posting that issue, sorry I did not find that until just now, but  this is exactly what I'm talking about. Good to see that the issue has been discussed last year. 

I read the thoughtful conversation over at issue #2355 and understand some of the pro's and con's of having a toggle-menu approach vs an approach to "generic map styles" that include POI filtering. I think either would be great, and if other apps allow for more granular approach then maybe an OSM "map style" approach makes sense for the sake of simplicity. I notice that, at least in the current iOS app, neither approach has been implemented yet. 

Glad the issue is on your radar, and grateful for the work of the volunteer maintainers - hope this makes it into some future release!

**Comment by dickorydock** [comment]
As commented above, this is related to this discussion from 2022: https://github.com/organicmaps/organicmaps/issues/2355

**Comment by williamdes** [comment]
I think this is a good feature request, when hiking or visiting foreign countries

**Reply by biodranik** [reply]
As we are working on a separate outdoor style, POI priorities can be lowered there in favor of street names. @pastk

**Reply by dickorydock** [reply]
Hello, I have tried the new outdoors hiking version. Is there a place somewhere that explains how the priorities work with this? Please see attached for a screenshot where you can still see many POIs labeled, yellow highlighted streets are all missing street names in this view. Is this expected behavior? @pastk 
![IMG_4507](https://github.com/organicmaps/organicmaps/assets/13089612/c9cb4cb8-7be9-4fa3-a8e4-18c9071e6fa5)

**Reply by pastk** [reply]
ATM outdoors style displays all the POIs present in the main style (and with the same priorities). With the difference that some outdoors-specific POIs visibilities were increased.

In the future we'll likely remove some POIs not related to Outdoors and it'll naturally give more space to the relevant stuff.

However note that emphasizing street names is not a direct goal of the Outdoors style (could be a side-effect though).

**Reply by biodranik** [reply]
@pastk is it easy to increase streets and paths texts priorities over POI priorities? Should we try that? Although displaying dots for POIs in this case would be great too.

**Reply by pastk** [reply]
Do you mean in the Outdoors style? Its not intended for that.
Or you meant somewhere else?

**Reply by biodranik** [reply]
Yes, in the outdoor style. When someone selects outdoors, shops and restaurants/cafes (and many other POIs) become less important. We can experiment with Outdoors without sacrificing the general style usability, and then, if necessary, make another layer/style for streets.

**Reply by pastk** [reply]
I'd argue that seeing grocery shops and places to eat (usually in smaller villages / towns on the way) is much more important than seeing street names.
Also its counter-intuitive for a user to switch to the Outdoors to see street names better..

If the sole point is in street names visibility then we have a vehicle style already which was intended and works pretty good for this case.

**Reply by biodranik** [reply]
You're right. Let's focus on adding a switch to enable vehicle style then and make it more usable )

**Comment by dickorydock** [comment]
Just came across @pastk's comment in [a different issue ](https://github.com/organicmaps/organicmaps/issues/2358)about searching for "?vlight" and that's basically just what I want here! It makes using the map much easier for me and I look forward to it being its own style. Thanks for adding that!
